src/compliance-engine-assessor/Main.cpp

```diff
@@ -0,0 +1,391 @@
+                    options.invalidArguments = true;
+                    break;
+                }
+
+                continue;
+            }
+
+            if (arg == "--format")
+            {
+                if (i + 1 < argc)
+                {
+                    std::string formatArg = argv[++i];
+                    std::transform(formatArg.begin(), formatArg.end(), formatArg.begin(), ::tolower);
+                    if (formatArg == "nested-list")
+                    {
+                        options.format = Format::NestedList;
+                    }
+                    else if (formatArg == "compact-list")
+                    {
+                        options.format = Format::CompactList;
+                    }
+                    else if (formatArg == "json")
+                    {
+                        options.format = Format::Json;
+                    }
+                    else if (formatArg == "mmi")
+                    {
+                        options.format = Format::Mmi;
+                    }
+                    else
+                    {
+                        std::cerr << "Invalid format: " << formatArg << "\n";
+                        options.command = Command::Help;
+                        options.invalidArguments = true;
+                        break;
+                    }
+                }
+                else
+                {
+                    std::cerr << "Missing argument for --format\n";
+                    options.command = Command::Help;
+                    options.invalidArguments = true;
+                    break;
+                }
+                continue;
+            }
+
+            if (arg == "audit")
+            {
+                options.command = Command::Audit;
+                continue;
+            }
+
+            if (arg == "remediate")
+            {
+                options.command = Command::Remediate;
+                continue;
+            }
+
+            if (Command::Undefined == options.command)
+            {
+                std::cerr << "Invalid command: " << arg << "\n";
+                options.command = Command::Help;
+                options.invalidArguments = true;
+                break;
+            }
+
+            if (options.input.empty())
+            {
+                options.input = arg;
+                continue;
+            }
+
+            std::cerr << "Unexpected argument: " << arg << "\n";
+            options.command = Command::Help;
+            options.invalidArguments = true;
+            break;
+        }
+
+        if (Command::Undefined == options.command)
+        {
+            std::cerr << "No command specified.\n";
+            options.command = Command::Help;
+            options.invalidArguments = true;
+        }
+
+        return options;
+    }
+
+    void PrintHelp()
+    {
+        std::cout << "Usage: assessor [--help] [--verbose] [--debug] {audit|remediate} [<input>]\n"
```

Comment:
>std::cout [](http://example.com/codeflow?start=8&length=9)

The assessor tool can link to the logging lib and use OSConfig tracing to log to its own file, and with that also get by default console routing when ran as a console tool. Would you consider this?